app/controllers/replies_controller.rb

```diff
@@ -297,19 +306,153 @@ def require_edit_permission
     redirect_to post_path(@reply.post)
   end
 
-  def preview(written)
-    @written = written
-    @post = @written.post
+  def get_multi_replies
+    # Get the multi replies stored in the page JSON
+    @multi_replies_params = JSON.parse(params.fetch(:multi_replies_json, "[]")).map do |reply_json|
+      permitted_params(ActionController::Parameters.new({ reply: reply_json }), [:id])
+    end
+    @multi_replies = @multi_replies_params.map do |reply_params|
+      Reply.new(reply_params).tap { |r| r.user = current_user }
+    end
+  end
+
+  def preview_reply(reply)
+    # Previewing a specific reply
+    @post = reply.post
+    @written = reply
     @written.user = current_user unless @written.user
     @audits = @written.id.present? ? { @written.id => @written.audits.count } : {}
+    @reply = @written # So that editor_setup shows the correct characters
+    preview_replies
+  end
 
+  def add_to_multi_reply(reply, reply_params)
+    # Adding a new reply to a multi reply
+    post_id = params[:reply][:post_id]
+    ReplyDraft.draft_for(post_id, current_user.id)&.destroy!
+
+    # Save NPC
+    if reply.character&.new_record?
+      if reply.character.save
+        flash[:success] = "Your new NPC has been persisted!"
+        params[:reply][:character_id] = reply.character.id
+        reply_params[:character_id] = reply.character.id
+        reply.character_id = reply.character.id
+      else
+        flash[:error] = "There was a problem persisting your new NPC."
+      end
+    end
```

Comment:
Terrifying!!! too many things to handle.

Reply:
😔

Reply:
Oh but i think you could
```suggestion
    # Save NPC
    if reply.character&.new_record?
      if reply.character.save
        flash[:success] = "Your new NPC has been persisted!"
        params[:reply][:character_id], reply_params[:character_id], reply.character_id = reply.character.id
      else
        flash[:error] = "There was a problem persisting your new NPC."
      end
    end
```
I'm not sure you *should* but I'm pretty sure that works

Reply:
Do you actually have to manually set `reply.character_id` here, rails doesn't handle that?

Reply:
It doesn't! I found it out the hard way!

I also think I don't want to do the thing you suggested if you don't have a super strong preference because it is less readable to me personally but I'm not married to mine

Reply:
If there were only two of them I might encourage that, but with three it starts being a Lot imo, feel free to go with your thing.